File: 2024/07/20/202. Happy Number/solution.py
class Solution(object):
    def isHappy(self, n):
        """
        :type n: int
        :rtype: bool
        """
        de_set=set()
        while n is not 1:
            n = sum(int(i)**2 for i in str(n))
            if n in de_set:
                return False
            de_set.add(n)
        return True

File: 2024/07/20/202. Happy Number/solution.cpp
class Solution {
public:
    bool isHappy(int n) {
        unordered_set<int> myset;
        while (n!=1 && myset.find(n)==myset.end()){
            myset.insert(n);
            n=help(n);
        }
        return n==1;
    }

private:
    int help(int n){
        int sum = 0;
        while (n > 0) {
            int digit = n % 10;
            sum += digit * digit;
            n /= 10;
        }
        return sum;
    }
};